config/crd/bases/rabbitmq.com_federations.yaml

```diff
@@ -74,17 +74,24 @@ spec:
                 type: integer
               trustUserId:
                 type: boolean
-              uri:
-                description: The AMQP URI(s) for the upstream. Required property.
-                type: string
+              uriSecret:
+                description: Secret contains the AMQP URI(s) for the upstream. The
+                  Secret must contain the key `uri` or operator will error. Required
+                  property.
+                properties:
+                  name:
+                    description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                      TODO: Add other useful fields. apiVersion, kind, uid?'
```

Comment:
Is the `TODO` still meant to be here?

Reply:
It comes from kubernetes api corev1.LocalObjectReference directly: https://github.com/kubernetes/api/blob/b0d9a0e45e6f6f7c5f18d76d950a4d19e41d1f50/core/v1/types.go#L5407